onnxruntime/core/optimizer/graph_transformer_mgr.cc

```diff
@@ -2,24 +2,48 @@
 // Licensed under the MIT License.
 
 #include "core/optimizer/graph_transformer_mgr.h"
+#include "core/optimizer/rule_based_graph_transformer.h"
 using namespace onnxruntime;
 using namespace ::onnxruntime::common;
 
 namespace onnxruntime {
 
-Status GraphTransformerManager::ApplyAll(Graph& graph) const {
+common::Status GraphTransformerManager::ApplyTransformers(Graph& graph, std::vector<std::string> providers, const TransformerLevel& level) const {
+  const auto& transformers = level_to_transformer_map_.find(level);
+  if (transformers == level_to_transformer_map_.end()) {
+    return Status::OK();
+  }
+
+  // sort the providers. This makes it easy to find intersection of requested providers for this run
+  // and registered providers for a transformer
+  std::sort(providers.begin(), providers.end());
+
```

Comment:
Are there scenarios where we want to run the provider specific transformers in the given order? e.g. if I have a TensorRT provider and a CUDA one, would I maybe want the TensorRT transformers to always run prior to the assumably lower level CUDA ones?

Reply:
Nope - order does not matter...

---
In reply to: [265379616](https://github.com/Microsoft/onnxruntime/pull/608#discussion_r265379616) [](ancestors = 265379616)

Reply:
In that case would it be better to take providers as a std::set instead of sorting vectors?

---
In reply to: [265380204](https://github.com/Microsoft/onnxruntime/pull/608#discussion_r265380204) [](ancestors = 265380204,265379616)